LoopKitUI/View Controllers/BasalScheduleTableViewController.swift

```diff
@@ -0,0 +1,399 @@
+                    cell.isEnabled = !isSyncInProgress
+                    cell.isLoading = isSyncInProgress
+                case let cell as BasalScheduleEntryTableViewCell:
+                    cell.isReadOnly = isReadOnly || isSyncInProgress
+                default:
+                    break
+                }
+            }
+
+            for item in navigationItem.rightBarButtonItems ?? [] {
+                item.isEnabled = !isSyncInProgress
+            }
+
+            navigationItem.hidesBackButton = isSyncInProgress
+        }
+    }
+
+    public var isScheduleValid: Bool {
+        for item in scheduleItems {
+            if !validate(value: item.value) {
+                return false
+            }
+        }
+        return scheduleItems.count <= maximumScheduleItemCount
+    }
+
+
+    // MARK: - UITableViewDataSource
+
+    private enum Section: Int {
+        case schedule
+        case sync
+    }
+
+    open override func numberOfSections(in tableView: UITableView) -> Int {
+        if syncSource != nil {
+            return 2
+        }
+
+        return 1
+    }
+
+    open override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch Section(rawValue: section)! {
+        case .schedule:
+            return scheduleItems.count
+        case .sync:
+            return 1
+        }
+    }
+
+    open override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        switch Section(rawValue: indexPath.section)! {
+        case .schedule:
+            let cell = tableView.dequeueReusableCell(withIdentifier: BasalScheduleEntryTableViewCell.className, for: indexPath) as! BasalScheduleEntryTableViewCell
+
+            let item = scheduleItems[indexPath.row]
+
+            cell.valueNumberFormatter.minimumFractionDigits = preferredValueFractionDigits()
+            cell.maximumBasalRatePerHour = maximumBasalRatePerHour
+            cell.minimumBasalRatePerHour = minimumBasalRatePerHour
+            cell.minimumRateIncrement = minimumRateIncrement
+            cell.unitString = unitDisplayString
+            cell.pickerInterval = minimumTimeInterval
+            cell.isReadOnly = isReadOnly || isSyncInProgress
+            cell.delegate = self
+
+            if indexPath.row > 0 {
+                let lastItem = scheduleItems[indexPath.row - 1]
+
+                cell.minimumStartTime = lastItem.startTime + minimumTimeInterval
+            }
+
+            if indexPath.row == 0 {
+                cell.maximumStartTime = 0
+            } else if indexPath.row < scheduleItems.endIndex - 1 {
+                let nextItem = scheduleItems[indexPath.row + 1]
+                cell.maximumStartTime = nextItem.startTime - minimumTimeInterval
+            } else {
+                cell.maximumStartTime = TimeInterval(hours: 24) - minimumTimeInterval
+            }
+
+            cell.value = item.value
```

Comment:
Do we need to pin `value` and `startTime` to min/max set above?

Reply:
[nit] ping

Reply:
I don't think so; value should be left at the user's value, and highlighted as invalid for them to change as they want.  An invalid startTime would indicate a bug somewhere in the code.